packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -1,193 +1,63 @@
-import React, {
-    useCallback,
-    useEffect,
-    useRef,
-    useState,
-    useMemo,
-    isValidElement,
-} from 'react';
-import { styled } from '@mui/material/styles';
-import Downshift, { DownshiftProps } from 'downshift';
+import * as React from 'react';
+import { isValidElement, useEffect, useMemo, useRef, useState } from 'react';
 import get from 'lodash/get';
-import classNames from 'classnames';
-import {
-    TextField,
-    InputAdornment,
-    IconButton,
-    InputProps,
-} from '@mui/material';
-import ClearIcon from '@mui/icons-material/Clear';
-import { TextFieldProps } from '@mui/material/TextField';
+import { Autocomplete, AutocompleteProps, TextField } from '@mui/material';
 import {
     useInput,
     FieldTitle,
     ChoicesInputProps,
     UseChoicesOptions,
-    mergeRefs,
     useSuggestions,
+    useTimeout,
     useTranslate,
     warning,
 } from 'ra-core';
-
-import { InputHelperText } from './InputHelperText';
-import { AutocompleteSuggestionList } from './AutocompleteSuggestionList';
```

Comment:
I assume that these are still required for AutocompleteArrayInput, and that's why you didn't delete them in this PR, right?

Reply:
Exactly